Signal/src/Models/TSMessageAdapaters/TSVideoAttachmentAdapter.m

```diff
@@ -259,10 +204,149 @@ - (UIView *)mediaView {
     return self.cachedMediaView;
 }
 
+- (UIView *)createVideoMediaView
+{
+    OWSAssert([self isVideo]);
+
+    CGSize size = [self mediaViewDisplaySize];
+
+    UIImageView *imageView = [[UIImageView alloc] initWithImage:self.image];
+    imageView.contentMode = UIViewContentModeScaleAspectFill;
+    imageView.frame = CGRectMake(0.0f, 0.0f, size.width, size.height);
+    imageView.clipsToBounds = YES;
+    [JSQMessagesMediaViewBubbleImageMasker applyBubbleImageMaskToMediaView:imageView
+                                                                isOutgoing:self.appliesMediaViewMaskAsOutgoing];
+    UIImage *img = [UIImage imageNamed:@"play_button"];
+    UIImageView *videoPlayButton = [[UIImageView alloc] initWithImage:img];
+    videoPlayButton.frame = CGRectMake((size.width / 2) - 18, (size.height / 2) - 18, 37, 37);
+    [imageView addSubview:videoPlayButton];
+
+    if (!_incoming) {
+        self.attachmentUploadView = [[AttachmentUploadView alloc] initWithAttachment:self.attachment
+                                                                           superview:imageView
+                                                             attachmentStateCallback:^(BOOL isAttachmentReady) {
+                                                                 videoPlayButton.hidden = !isAttachmentReady;
+                                                             }];
+    }
+
+    return imageView;
+}
+
+- (BOOL)isVoiceMessage
+{
+    OWSAssert([self isAudio]);
+
+    return (self.attachment.isVoiceMessage || self.attachment.filename.length < 1);
```

Comment:
nit: maybe a comment about the legacy/arcane nature of `self.attachment.filename.length < 1`